### PR TITLE
Enhance .travis.yml

### PR DESCRIPTION
**Pull request by wkentaro**

### CODE REVIEW

**Comment by paulbovbel** [comment]
Does the docker script end up rebuilding the same artifacts several times? In that case I imagine ccache would be useful.

Please try to provided at least a minimal description with your pull requests.

**Comment by wkentaro** [comment]
- sudo: false: for faster booting on Travis
- ccache: in the testing script, it build twice with/without --install option, and also ccache can be cached with `cache: ccache` option in .travis.yml

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,11 @@
-sudo: required
+sudo: false
 dist: trusty
 language: generic
+services:
+  - docker
+cache:
```
Comment (paulbovbel) [inline .travis.yml]: It looks like setting:

`cache: ccache`

Will automatically install/add to path ccache, and cache the correct directory.

https://docs.travis-ci.com/user/caching/

**Review by paulbovbel — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (paulbovbel) [reply .travis.yml]: EDIT: ah but perhaps it won't pass that to the docker container. So can just change:

```
cache:
  directories:
    - $HOME/.ccache
```

to `cache: ccache`

Comment (wkentaro) [reply .travis.yml]: Ok, I updated.

**Review by paulbovbel — APPROVED** [review]

**Pull request merged**